src/flat.js

```diff
@@ -0,0 +1,8 @@
+export function flat(depth = 1) {
```

Comment:
How is `depth` used in this function? Is depth supported? Should tests be added for this?

Reply:
My `depth` implementation was broken. 🤦‍♂️ It passed tests only because the native `flat()` was being used during recursion. I just pushed a fix for that.

Reply:
I have added some tests for default (`depth=1`) and depth arg 2. Lmk if there's more you think needs to be covered.